src/app/business/delfin/storages/storages.component.html

```diff
@@ -0,0 +1,528 @@
+                    <ng-template let-storage pTemplate="rowexpansion">
+                        <div class="details-basic-info">
+                            <div class="ui-grid ui-grid-responsive ui-grid-pad ui-fluid">
+                                <div class="ui-grid-col-12">
+                                    <div class="ui-grid-row details-basic-item-class">
+                                        <div class="ui-grid-col-2">
+                                            {{label.name}}:
+                                        </div>
+                                        <div class="ui-grid-col-2 details-basic-item-value-color" *ngIf="storage">
+                                            {{storage.name}}
+                                        </div>
+                                        <div class="ui-grid-col-2">
+                                            {{label.status}}:
+                                        </div>
+                                        <div class="ui-grid-col-2 details-basic-item-value-color" *ngIf="storage">
+                                            <span class="storage-status-icon" [ngClass]="{normal: storage.status=='normal', abnormal: storage.status=='abnormal' || storage.status=='offline'}"><i class="fa fa-circle"></i></span>
+                                            <span *ngIf="storage.status=='normal'">Normal</span>
+                                            <span *ngIf="storage.status=='abnormal'">Abnormal</span>
+                                            <span *ngIf="storage.status=='offline'">Offline</span>
+                                        </div>
+                                        <div class="ui-grid-col-2">
+                                            {{label.created_at}}:
+                                        </div>
+                                        <div class="ui-grid-col-3 details-basic-item-value-color" *ngIf="storage">
+                                            {{storage.created_at ? (storage.created_at | date:'long') : '--'}}
+                                        </div>
+                                    
+                                    </div>
+                                    <div class="ui-grid-row details-basic-item-class">
+                                        <div class="ui-grid-col-2">
+                                            {{label.id}}:
+                                        </div>
+                                        <div class="ui-grid-col-2 details-basic-item-value-color" *ngIf="storage">
+                                            {{storage.id | slice:0:20}}
+                                            <a *ngIf="storage.id.length > 20" pTooltip="{{storage.id}}" tooltipPosition="top">
+                                                ...
+                                            </a>
+                                        </div>
+                                        <div class="ui-grid-col-2">
```

Comment:
Is there any possibility of adding list of properties and reading them in a loop? Else, everytime a field is added, the code here need to be changed and the this file keep growing.

Reply:
This may not be possible as each of the fields is a different type and there is no type identifier being returned in the API response. There are some fields which require some calculations and checks. i will explore any possibilities.

Reply:
ok